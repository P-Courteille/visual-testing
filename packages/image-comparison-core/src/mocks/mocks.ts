import type { BeforeScreenshotOptions } from '../helpers/beforeScreenshot.interfaces.js'

export const BEFORE_SCREENSHOT_OPTIONS: BeforeScreenshotOptions = {
    instanceData: {
        appName: 'chrome-app',
        browserName: 'chrome',
        browserVersion: '75.0.1',
        deviceName: '',
        devicePixelRatio: 1,
        initialDevicePixelRatio: 1,
        deviceRectangles: {
            statusBar: { y: 0, x: 0, width: 0, height: 0 },
            homeBar: { y: 0, x: 0, width: 0, height: 0 },
            screenSize: { height: 0, width: 0 },
            statusBarAndAddressBar: { y: 0, x: 0, width: 0, height: 0 },
            viewport: { y: 0, x: 0, width: 0, height: 0 },
            bottomBar: { y: 0, x: 0, width: 0, height: 0 },
            leftSidePadding: { y: 0, x: 0, width: 0, height: 0 },
            rightSidePadding: { y: 0, x: 0, width: 0, height: 0 },
        },
        isAndroid: false,
        isIOS: false,
        isMobile: false,
        logName: 'chrome-latest',
        name: 'chrome-name',
        nativeWebScreenshot: true,
        platformName: 'Windows 10',
        platformVersion: '1234',
    },
    addressBarShadowPadding: 6,
    disableBlinkingCursor: true,
    disableCSSAnimation: true,
    enableLayoutTesting: false,
    noScrollBars: true,
    toolBarShadowPadding: 6,
    hideElements: [<HTMLElement>(<any>'<div></div>')],
    removeElements: [<HTMLElement>(<any>'<div></div>')],
    waitForFontsLoaded: true,
}
export const CONFIGURABLE = {
    writable: true,
    configurable: true,
}
export const NAVIGATOR_APP_VERSIONS = {
    ANDROID: {
        7: '5.0 (Linux; Android 7.1.1; Android SDK built for x86 Build/NYC) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/55.0.2883.91 Mobile Safari/537.3',
        8: '5.0 (Linux; Android 8.1; Android SDK built for x86 Build/NYC) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/55.0.2883.91 Mobile Safari/537.3',
        9: '5.0 (Linux; Android 9; Android SDK built for x86 Build/NYC) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/55.0.2883.91 Mobile Safari/537.3',
        10: '5.0 (Linux; Android 10; Android SDK built for x86 Build/NYC) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/55.0.2883.91 Mobile Safari/537.3',
        11: '5.0 (Linux; Android 11; Android SDK built for x86 Build/NYC) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/55.0.2883.91 Mobile Safari/537.3',
    },
    IOS: {
        10: '5.0 (iPhone; CPU iPhone OS 10_2 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/10.0 Mobile/15E148 Safari/604.1',
        11: '5.0 (iPhone; CPU iPhone OS 11_4 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/11.0 Mobile/15E148 Safari/604.1',
        12: '5.0 (iPhone; CPU iPhone OS 12_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/12.0 Mobile/15E148 Safari/604.1',
        13: '5.0 (iPhone; CPU iPhone OS 13_3 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/13.0 Mobile/15E148 Safari/604.1',
        14: '5.0 (iPhone; CPU iPhone OS 14_0 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.0 Mobile/15E148 Safari/604.1',
        15: '5.0 (iPhone; CPU iPhone OS 15_0 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/15.0 Mobile/15E148 Safari/604.1',
    },
    IPADOS: {
        13: 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/13.0.1 Safari/605.1.15',
        14: 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.0 Safari/605.1.15',
        15: 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/15.0 Safari/605.1.15',
    },
}
export const ANDROID_DEVICES = {
    NEXUS_5X: {
        height: 732,
        width: 412,
        innerHeight: 604,
        innerWidth: 412,
    },
    NEXUS_5X_INNER_HEIGHT: {
        height: 732,
        width: 412,
        innerHeight: 800,
        innerWidth: 412,
    },
    TABLET_WIDTH: {
        height: 768,
        width: 1024,
        innerHeight: 604,
        innerWidth: 412,
    },
}
export const IOS_DEVICES = {
    IPHONE: {
        height: 667,
        width: 375,
        innerHeight: 553,
        innerWidth: 375,
        scrollWidth: 375,
        sideBar: 0,
    },
    IPHONE_X: {
        height: 812,
        width: 375,
        innerHeight: 635,
        innerWidth: 375,
        scrollWidth: 375,
        sideBar: 0,
    },
    IPHONE_HEIGHT: {
        height: 896,
        width: 1024,
        innerHeight: 719,
        innerWidth: 414,
        scrollWidth: 414,
        sideBar: 0,
    },
    IPHONE_11: {
        height: 896,
        width: 375,
        innerHeight: 635,
        innerWidth: 375,
        scrollWidth: 375,
        sideBar: 0,
    },
    IPAD: {
        height: 1366,
        width: 1024,
        innerHeight: 1292,
        innerWidth: 1024,
        scrollWidth: 1024,
        sideBar: 0,
    },
    IPAD_LANDSCAPE: {
        height: 1366,
        width: 1024,
        innerHeight: 746,
        innerWidth: 1046,
        scrollWidth: 1046,
        sideBar: 320,
    },
    IPAD_BIG_SIZE: {
        height: 5432,
        width: 9876,
        innerHeight: 5324,
        innerWidth: 9768,
        scrollWidth: 9768,
        sideBar: 108,
    },
    IPAD_PRO_LANDSCAPE: {
        height: 1366,
        width: 1024,
        innerHeight: 954,
        innerWidth: 1046,
        scrollWidth: 1046,
        sideBar: 320,
    },
}

export const BASE_CHECK_OPTIONS = {
    wic: {
        addressBarShadowPadding: 6,
        autoElementScroll: true,
        addIOSBezelCorners: false,
        autoSaveBaseline: false,
        clearFolder: false,
        userBasedFullPageScreenshot: false,
        enableLegacyScreenshotMethod: false,
        formatImageName: '{tag}-{logName}-{width}x{height}-dpr-{dpr}',
        isHybridApp: false,
        savePerInstance: true,
        toolBarShadowPadding: 6,
        disableBlinkingCursor: false,
        disableCSSAnimation: false,
        enableLayoutTesting: false,
        fullPageScrollTimeout: 1500,
        hideScrollBars: true,
        waitForFontsLoaded: true,
        compareOptions: {
            ignoreAlpha: false,
            ignoreAntialiasing: false,
            ignoreColors: false,
            ignoreLess: false,
            ignoreNothing: false,
            rawMisMatchPercentage: false,
            returnAllCompareData: false,
            saveAboveTolerance: 0,
            scaleImagesToSameSize: false,
            blockOutSideBar: false,
            blockOutStatusBar: false,
            blockOutToolBar: false,
            createJsonReportFiles: false,
            diffPixelBoundingBoxProximity: 5,
        },
        tabbableOptions: {
            circle: {
                backgroundColor: 'rgba(255, 0, 0, 0.4)',
                borderColor: 'rgba(255, 0, 0, 1)',
                borderWidth: 1,
                fontColor: 'rgba(0, 0, 0, 1)',
                fontFamily: 'Arial',
                fontSize: 10,
                size: 10,
            },
            line: {
                color: 'rgba(255, 0, 0, 1)',
                width: 1,
            },
        }
    },
    instanceData: {
        appName: 'TestApp',
        browserName: 'Chrome',
        browserVersion: '118.0.0.0',
        deviceName: 'iPhone 14',
        devicePixelRatio: 2,
        deviceRectangles: {
            bottomBar: { y: 800, x: 0, width: 390, height: 0 },
            homeBar: { x: 0, y: 780, width: 390, height: 34 },
            leftSidePadding: { y: 0, x: 0, width: 0, height: 0 },
            rightSidePadding: { y: 0, x: 0, width: 0, height: 0 },
            screenSize: { height: 844, width: 390 },
            statusBar: { x: 0, y: 0, width: 390, height: 47 },
            statusBarAndAddressBar: { y: 0, x: 0, width: 390, height: 47 },
            viewport: { y: 47, x: 0, width: 390, height: 733 }
        },
        initialDevicePixelRatio: 2,
        isAndroid: false,
        isIOS: true,
        isMobile: true,
        logName: 'test-log',
        name: 'test-device',
        nativeWebScreenshot: false,
        platformName: 'iOS',
        platformVersion: '17.0'
    },
    folders: {
        actualFolder: '/test/actual',
        baselineFolder: '/test/baseline',
        diffFolder: '/test/diff'
    },
    testContext: {
        commandName: 'checkScreen',
        framework: 'vitest',
        parent: 'test suite',
        title: 'test title',
        tag: 'test-tag',
        instanceData: {
            browser: { name: 'Chrome', version: '118.0.0.0' },
            deviceName: 'iPhone 14',
            platform: { name: 'iOS', version: '17.0' },
            app: 'TestApp',
            isMobile: true,
            isAndroid: false,
            isIOS: true
        }
    }
}
