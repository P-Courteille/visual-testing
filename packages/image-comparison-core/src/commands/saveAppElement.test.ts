import { describe, it, expect, vi, beforeEach, afterEach } from 'vitest'
import saveAppElement from './saveAppElement.js'
import type { InternalSaveElementMethodOptions } from './save.interfaces.js'
import { BASE_CHECK_OPTIONS } from '../mocks/mocks.js'

vi.mock('../methods/images.js', () => ({
    takeBase64ElementScreenshot: vi.fn().mockResolvedValue('base64-image-data')
}))
vi.mock('../helpers/afterScreenshot.js', () => ({
    default: vi.fn().mockResolvedValue({
        devicePixelRatio: 2,
        fileName: 'test-element.png'
    })
}))

describe('saveAppElement', () => {
    let takeBase64ElementScreenshotSpy: ReturnType<typeof vi.fn>
    let afterScreenshotSpy: ReturnType<typeof vi.fn>

    const baseOptions: InternalSaveElementMethodOptions = {
        browserInstance: { isAndroid: false, isMobile: false } as any,
        element: { elementId: 'test-element' } as any,
        folders: BASE_CHECK_OPTIONS.folders,
        instanceData: BASE_CHECK_OPTIONS.instanceData,
        isNativeContext: true,
        saveElementOptions: {
            wic: BASE_CHECK_OPTIONS.wic,
            method: {
                resizeDimensions: undefined
            }
        },
        tag: 'test-element'
    }

    beforeEach(async () => {
        const { takeBase64ElementScreenshot } = await import('../methods/images.js')
        const afterScreenshot = (await import('../helpers/afterScreenshot.js')).default

        takeBase64ElementScreenshotSpy = vi.mocked(takeBase64ElementScreenshot)
        afterScreenshotSpy = vi.mocked(afterScreenshot)
    })

    afterEach(() => {
        vi.clearAllMocks()
    })

    it('should execute saveAppElement with basic options', async () => {
        const result = await saveAppElement(baseOptions)

        expect(result).toMatchSnapshot()
        expect(takeBase64ElementScreenshotSpy.mock.calls[0]).toMatchSnapshot()
        expect(afterScreenshotSpy.mock.calls[0]).toMatchSnapshot()
    })

    it('should handle custom resize dimensions', async () => {
        const options = {
            ...baseOptions,
            saveElementOptions: {
                ...baseOptions.saveElementOptions,
                method: {
                    resizeDimensions: {
                        width: 100,
                        height: 100,
                        top: 0,
                        left: 0,
                        right: 0,
                        bottom: 0
                    }
                }
            }
        }

        await saveAppElement(options)

        expect(takeBase64ElementScreenshotSpy.mock.calls[0]).toMatchSnapshot()
    })

    it('should handle iOS device correctly', async () => {
        const options = {
            ...baseOptions,
            instanceData: {
                ...baseOptions.instanceData,
                deviceName: 'iPhone 14',
                isAndroid: false,
                isIOS: true,
                platformName: 'iOS',
                platformVersion: '17.0'
            }
        }

        await saveAppElement(options)

        expect(takeBase64ElementScreenshotSpy.mock.calls[0]).toMatchSnapshot()
        expect(afterScreenshotSpy.mock.calls[0]).toMatchSnapshot()
    })

    it('should handle Android device correctly', async () => {
        const options = {
            ...baseOptions,
            instanceData: {
                ...baseOptions.instanceData,
                deviceName: 'Pixel 4',
                isAndroid: true,
                isIOS: false,
                platformName: 'Android',
                platformVersion: '11.0'
            }
        }

        await saveAppElement(options)

        expect(takeBase64ElementScreenshotSpy.mock.calls[0]).toMatchSnapshot()
        expect(afterScreenshotSpy.mock.calls[0]).toMatchSnapshot()
    })

    it('should handle non-native context correctly', async () => {
        const options = {
            ...baseOptions,
            isNativeContext: false
        }

        await saveAppElement(options)

        expect(afterScreenshotSpy.mock.calls[0]).toMatchSnapshot()
    })

    it('should handle custom format image name', async () => {
        const options = {
            ...baseOptions,
            saveElementOptions: {
                ...baseOptions.saveElementOptions,
                wic: {
                    ...baseOptions.saveElementOptions.wic,
                    formatImageName: '{tag}-{browserName}-{deviceName}'
                }
            }
        }

        await saveAppElement(options)

        expect(afterScreenshotSpy.mock.calls[0]).toMatchSnapshot()
    })

    it('should handle savePerInstance option', async () => {
        const options = {
            ...baseOptions,
            saveElementOptions: {
                ...baseOptions.saveElementOptions,
                wic: {
                    ...baseOptions.saveElementOptions.wic,
                    savePerInstance: false
                }
            }
        }

        await saveAppElement(options)

        expect(afterScreenshotSpy.mock.calls[0]).toMatchSnapshot()
    })

    it('should handle custom screen size', async () => {
        const options = {
            ...baseOptions,
            instanceData: {
                ...baseOptions.instanceData,
                deviceRectangles: {
                    ...baseOptions.instanceData.deviceRectangles,
                    screenSize: {
                        width: 1920,
                        height: 1080
                    }
                }
            }
        }

        await saveAppElement(options)

        expect(afterScreenshotSpy.mock.calls[0]).toMatchSnapshot()
    })
})
