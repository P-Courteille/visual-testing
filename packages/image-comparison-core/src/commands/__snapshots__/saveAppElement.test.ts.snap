// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`saveAppElement > should execute saveAppElement with basic options 1`] = `
{
  "devicePixelRatio": 2,
  "fileName": "test-element.png",
}
`;

exports[`saveAppElement > should execute saveAppElement with basic options 2`] = `
[
  {
    "browserInstance": {
      "isAndroid": false,
      "isMobile": false,
    },
    "devicePixelRatio": 2,
    "element": {
      "elementId": "test-element",
    },
    "isIOS": true,
    "resizeDimensions": {
      "bottom": 0,
      "left": 0,
      "right": 0,
      "top": 0,
    },
  },
]
`;

exports[`saveAppElement > should execute saveAppElement with basic options 3`] = `
[
  {
    "isAndroid": false,
    "isMobile": false,
  },
  {
    "actualFolder": "/test/actual",
    "base64Image": "base64-image-data",
    "fileName": {
      "browserName": "Chrome",
      "browserVersion": "118.0.0.0",
      "deviceName": "iPhone 14",
      "devicePixelRatio": 2,
      "formatImageName": "{tag}-{logName}-{width}x{height}-dpr-{dpr}",
      "isMobile": true,
      "isTestInBrowser": false,
      "logName": "test-log",
      "name": "",
      "platformName": "iOS",
      "platformVersion": "17.0",
      "screenHeight": 844,
      "screenWidth": 390,
      "tag": "test-element",
    },
    "filePath": {
      "browserName": "Chrome",
      "deviceName": "iPhone 14",
      "isMobile": true,
      "savePerInstance": true,
    },
    "isLandscape": false,
    "isNativeContext": true,
    "platformName": "iOS",
  },
]
`;

exports[`saveAppElement > should handle Android device correctly 1`] = `
[
  {
    "browserInstance": {
      "isAndroid": false,
      "isMobile": false,
    },
    "devicePixelRatio": 2,
    "element": {
      "elementId": "test-element",
    },
    "isIOS": false,
    "resizeDimensions": {
      "bottom": 0,
      "left": 0,
      "right": 0,
      "top": 0,
    },
  },
]
`;

exports[`saveAppElement > should handle Android device correctly 2`] = `
[
  {
    "isAndroid": false,
    "isMobile": false,
  },
  {
    "actualFolder": "/test/actual",
    "base64Image": "base64-image-data",
    "fileName": {
      "browserName": "Chrome",
      "browserVersion": "118.0.0.0",
      "deviceName": "Pixel 4",
      "devicePixelRatio": 2,
      "formatImageName": "{tag}-{logName}-{width}x{height}-dpr-{dpr}",
      "isMobile": true,
      "isTestInBrowser": false,
      "logName": "test-log",
      "name": "",
      "platformName": "Android",
      "platformVersion": "11.0",
      "screenHeight": 844,
      "screenWidth": 390,
      "tag": "test-element",
    },
    "filePath": {
      "browserName": "Chrome",
      "deviceName": "Pixel 4",
      "isMobile": true,
      "savePerInstance": true,
    },
    "isLandscape": false,
    "isNativeContext": true,
    "platformName": "Android",
  },
]
`;

exports[`saveAppElement > should handle custom format image name 1`] = `
[
  {
    "isAndroid": false,
    "isMobile": false,
  },
  {
    "actualFolder": "/test/actual",
    "base64Image": "base64-image-data",
    "fileName": {
      "browserName": "Chrome",
      "browserVersion": "118.0.0.0",
      "deviceName": "iPhone 14",
      "devicePixelRatio": 2,
      "formatImageName": "{tag}-{browserName}-{deviceName}",
      "isMobile": true,
      "isTestInBrowser": false,
      "logName": "test-log",
      "name": "",
      "platformName": "iOS",
      "platformVersion": "17.0",
      "screenHeight": 844,
      "screenWidth": 390,
      "tag": "test-element",
    },
    "filePath": {
      "browserName": "Chrome",
      "deviceName": "iPhone 14",
      "isMobile": true,
      "savePerInstance": true,
    },
    "isLandscape": false,
    "isNativeContext": true,
    "platformName": "iOS",
  },
]
`;

exports[`saveAppElement > should handle custom resize dimensions 1`] = `
[
  {
    "browserInstance": {
      "isAndroid": false,
      "isMobile": false,
    },
    "devicePixelRatio": 2,
    "element": {
      "elementId": "test-element",
    },
    "isIOS": true,
    "resizeDimensions": {
      "bottom": 0,
      "height": 100,
      "left": 0,
      "right": 0,
      "top": 0,
      "width": 100,
    },
  },
]
`;

exports[`saveAppElement > should handle custom screen size 1`] = `
[
  {
    "isAndroid": false,
    "isMobile": false,
  },
  {
    "actualFolder": "/test/actual",
    "base64Image": "base64-image-data",
    "fileName": {
      "browserName": "Chrome",
      "browserVersion": "118.0.0.0",
      "deviceName": "iPhone 14",
      "devicePixelRatio": 2,
      "formatImageName": "{tag}-{logName}-{width}x{height}-dpr-{dpr}",
      "isMobile": true,
      "isTestInBrowser": false,
      "logName": "test-log",
      "name": "",
      "platformName": "iOS",
      "platformVersion": "17.0",
      "screenHeight": 1080,
      "screenWidth": 1920,
      "tag": "test-element",
    },
    "filePath": {
      "browserName": "Chrome",
      "deviceName": "iPhone 14",
      "isMobile": true,
      "savePerInstance": true,
    },
    "isLandscape": false,
    "isNativeContext": true,
    "platformName": "iOS",
  },
]
`;

exports[`saveAppElement > should handle iOS device correctly 1`] = `
[
  {
    "browserInstance": {
      "isAndroid": false,
      "isMobile": false,
    },
    "devicePixelRatio": 2,
    "element": {
      "elementId": "test-element",
    },
    "isIOS": true,
    "resizeDimensions": {
      "bottom": 0,
      "left": 0,
      "right": 0,
      "top": 0,
    },
  },
]
`;

exports[`saveAppElement > should handle iOS device correctly 2`] = `
[
  {
    "isAndroid": false,
    "isMobile": false,
  },
  {
    "actualFolder": "/test/actual",
    "base64Image": "base64-image-data",
    "fileName": {
      "browserName": "Chrome",
      "browserVersion": "118.0.0.0",
      "deviceName": "iPhone 14",
      "devicePixelRatio": 2,
      "formatImageName": "{tag}-{logName}-{width}x{height}-dpr-{dpr}",
      "isMobile": true,
      "isTestInBrowser": false,
      "logName": "test-log",
      "name": "",
      "platformName": "iOS",
      "platformVersion": "17.0",
      "screenHeight": 844,
      "screenWidth": 390,
      "tag": "test-element",
    },
    "filePath": {
      "browserName": "Chrome",
      "deviceName": "iPhone 14",
      "isMobile": true,
      "savePerInstance": true,
    },
    "isLandscape": false,
    "isNativeContext": true,
    "platformName": "iOS",
  },
]
`;

exports[`saveAppElement > should handle non-native context correctly 1`] = `
[
  {
    "isAndroid": false,
    "isMobile": false,
  },
  {
    "actualFolder": "/test/actual",
    "base64Image": "base64-image-data",
    "fileName": {
      "browserName": "Chrome",
      "browserVersion": "118.0.0.0",
      "deviceName": "iPhone 14",
      "devicePixelRatio": 2,
      "formatImageName": "{tag}-{logName}-{width}x{height}-dpr-{dpr}",
      "isMobile": true,
      "isTestInBrowser": true,
      "logName": "test-log",
      "name": "",
      "platformName": "iOS",
      "platformVersion": "17.0",
      "screenHeight": 844,
      "screenWidth": 390,
      "tag": "test-element",
    },
    "filePath": {
      "browserName": "Chrome",
      "deviceName": "iPhone 14",
      "isMobile": true,
      "savePerInstance": true,
    },
    "isLandscape": false,
    "isNativeContext": false,
    "platformName": "iOS",
  },
]
`;

exports[`saveAppElement > should handle savePerInstance option 1`] = `
[
  {
    "isAndroid": false,
    "isMobile": false,
  },
  {
    "actualFolder": "/test/actual",
    "base64Image": "base64-image-data",
    "fileName": {
      "browserName": "Chrome",
      "browserVersion": "118.0.0.0",
      "deviceName": "iPhone 14",
      "devicePixelRatio": 2,
      "formatImageName": "{tag}-{logName}-{width}x{height}-dpr-{dpr}",
      "isMobile": true,
      "isTestInBrowser": false,
      "logName": "test-log",
      "name": "",
      "platformName": "iOS",
      "platformVersion": "17.0",
      "screenHeight": 844,
      "screenWidth": 390,
      "tag": "test-element",
    },
    "filePath": {
      "browserName": "Chrome",
      "deviceName": "iPhone 14",
      "isMobile": true,
      "savePerInstance": false,
    },
    "isLandscape": false,
    "isNativeContext": true,
    "platformName": "iOS",
  },
]
`;
